Indicators/LeastSquaresMovingAverage.cs

```diff
@@ -63,23 +75,31 @@ public LeastSquaresMovingAverage(int period)
         /// </returns>
         protected override decimal ComputeNextValue(IReadOnlyWindow<IndicatorDataPoint> window, IndicatorDataPoint input)
         {
-            // Until the windows is ready, the indicator returns the input value.
-            decimal output = input;
-            if (IsReady)
-            {
-                // Sort the windows by time, convert the observations ton double and transform it to a double array
-                double[] series = window
-                    .OrderBy(i => i.Time)
-                    .Select(i => Convert.ToDouble(i.Value))
-                    .ToArray<double>();
-                // Fit OLS
-                Tuple<double, double> ols = Fit.Line(x: t, y: series);
-                var alfa = (decimal)ols.Item1;
-                var beta = (decimal)ols.Item2;
-                // Make the projection.
-                output = alfa + beta * (Period);
-            }
-            return output;
+            // Until the window is ready, the indicator returns the input value.
+            if (!IsReady) return input;
+
+            // Sort the window by time, convert the observations to double and transform it to an array
+            var series = window
+                .OrderBy(i => i.Time)
+                .Select(i => Convert.ToDouble(i.Value))
+                .ToArray();
+            // Fit OLS
+            var ols = Fit.Line(x: t, y: series);
+            Intercept.Update(input.Time, (decimal)ols.Item1);
+            Slope.Update(input.Time, (decimal)ols.Item2);
+
+            // Calculate the fitted value corresponding to the input
+            return Intercept + Slope * Period;
```

Comment:
Variable renames are nice in their own commits. Ideally,  any change that has no effect on behavior should be in its own commit. This includes doc changes,  white space changes, renames and mechanical refactorings. This helps the diff engine during merges and helps the poor sap that needs to look through the history to find out how something for some way... It's much better to be able to completely ignore the renames in the diff and instead look at a change that is primarily changing behavior, vs a commit that changes behaviors and has renames and refactorings, ect...